### ISSUE TITLE
LLVM codegen compiles with failing assertions

### ISSUE THREAD

**Issue by ehsanmok** [dmlc/tvm#1117]
Hi

I'm on Ubuntu 16.04, LLVM 6.0, gcc 5.4, Python 3.6 Anaconda
Reproducing example:

```
import mxnet as mx
import tvm  # 0.2.0
import nnvm  # 0.8.0
from mxnet.gluon.model_zoo import vision

resnet18 = vision.get_resnet(version=2, num_layers=18, pretrained=True, ctx=mx.cpu())
resnet18.hybridize()  # with or without hybridize fails!

sym, params = nnvm.frontend.from_mxnet(resnet18)
sym = nnvm.sym.softmax(sym)
target = 'llvm -target=arm64-linux-android'
shape_dict = {'data': (1, 3, 224, 224)}
with nnvm.compiler.build_config(opt_level=0):
    graph, lib, param = nnvm.compiler.build(sym, target, shape=shape_dict,  params=params, target_host=None, dtype='float32')

print(lib.get_source())
```
outputs

```
........
@__TVMAPISetLastError = linkonce dllexport local_unnamed_addr global void (i8*)* null, align 8\n@__TVMBackendParallelLaunch = linkonce dllexport local_unnamed_addr global i32 (i32 (i32, %0*, i8*)*, i8*, i32)* null, align 8\n@.str = private constant [72 x i8] c"Assert fail: (num_args == 2), fuse___add_scalar__: num_args should be 2\\00", align 1\n@.str.1 = private constant [87 x i8] c"Assert fail: (1 == int32(arg0.strides[0])), arg0.strides: expected to be compact array\\00", align 1\n@.str.2 = private constant [87 x i8] c"Assert fail: (1 == int32(arg1.strides[0])), arg1.strides: expected to be compact array\\00", align 1\n@.str.3 = private constant [124 x i8] c"Assert fail: (((arg0.code == 3) || (arg0.code == 7)) || (arg0.code == 4)), fuse___add_scalar__: Expect arg[0] to be pointer\\00", align 1\n@.str.4 = private constant [124 x i8] c"Assert fail: (((arg1.code == 3) || (arg1.code == 7)) || (arg1.code == 4)), fuse___add_scalar__: Expect arg[1] to be pointer\\00", align 1\n@.str.5 = private constant [55 x i8] c"Assert fail: (dev_type == 1), device_type need to be 1\\00", align 1\n@.str.6 = private constant [81 x i8] c"Assert fail: (1 == tvm_struct_get(arg0, 0, 4)), arg0.ndim is expected to equal 1\\00", align 1\n@.str.7 = private constant [186 x i8] c"Assert fail: (((tvm_struct_get(arg0, 0, 5) == (uint8)2) && (tvm_struct_get(arg0, 0, 6) == (uint8)32)) && (tvm_struct_get(arg0, 0, 7) == (uint16)1)), arg0.dtype is expected to be float32\\00", align 1\n@.str.8 = private constant [95 x i8] c"Assert fail: (int32(arg0.shape[0]) == 3), Argument arg0.shape[0] has an unsatisfied constraint\\00", align 1\n@.str.9 = private constant [112 x i8] c"Assert fail: (tvm_struct_get(arg0, 0, 8) == (uint64)0), Argument arg0.byte_offset has an unsatisfied constraint\\00", align 1\n@.str.10 = private constant [81 x i8] c"Assert fail: (1 == tvm_struct_get(arg1, 0, 4)), arg1.ndim is expected to equal 1\\00", align 1\n@.str.11 = private constant [186 x i8] c"Assert fail: (((tvm_struct_get(arg1, 0, 5) == (uint8)2) && (tvm_struct_get(arg1, 0, 6) == (uint8)32)) && (tvm_struct_get(arg1, 0, 7) == (uint16)1)), arg1.dtype is expected to be float3
....
```

with a lot of fails though. Any idea?

**Comment by tqchen** [comment]
these asserts are runtime asserts that do not indicate compilation failure, but is the error message that the code will raise when these asserts are not met during runtime

**Comment by ehsanmok** [comment]
I see! thanks for the clarification. However, I suspect that batchNorm (used in ResNetV2) is not yet supported in tvm? that's why the android app crashes when loading the model!

**Comment by tqchen** [comment]
it is supported, and will get optimized away by translating that to scale and shift, but you need to set opt_level to be bigger than 1 to enable that

**Comment by ehsanmok** [comment]
Right! thanks, it works now. I mistakenly added `softmax` which caused the crash.
